### DISCUSSION TITLE
Comparison with different micro frontend frameworks

### DISCUSSION THREAD

**Discussion by paddomanno** [SAP/luigi#3502] [General]
Hello SAP/Luigi community!

I'm excited to start my journey into the world of micro frontends and the various frameworks that support this architecture. As I delve into this topic, I've come across the Luigi framework, and I'm keen to understand how it differentiates itself from other similar frameworks like Single-SPA and Piral.

My goal with this discussion is to explore the reasons behind the development of Luigi, what unique features it brings to the table, and what it enables you to build. I would also like to understand if there are any potential drawbacks or limitations associated with Luigi.

Here are some specific questions I have in mind, but feel free to share any insights you have:

- Why Luigi? - What were the driving factors that led to the development of Luigi as a micro frontend framework? Were there specific needs or gaps in the existing solutions that prompted its creation?
- Features and Capabilities - What are the standout features of Luigi that set it apart from alternatives like Single-SPA and Piral? How does it enhance the micro frontend development process?
- Use Cases - Could you share real-world use cases where SAP/Luigi excels? What kind of projects or scenarios is it best suited for?
- Drawbacks and Limitations - No framework is perfect. Are there any drawbacks or limitations that developers should be aware of when using Luigi? How does it compare in this regard to other frameworks?

I'm really looking forward to learning from your experiences and insights.

**Comment by alexandra-simeonova** [comment]
Hello! Thank you for your question and your interest in Luigi. Here are some key points about the framework:

* Why Luigi? - Luigi was developed in order to make micro frontend adoption easier, especially in the context of business applications. It is technology-agnostic and meant to work with different frontend frameworks like Angular, React, or others. It is specifically tailored to work with SAP applications, which are usually large-scale enterprise systems. 

* Features and Capabilities - Luigi offers several main features, such as an L-shaped navigation out-of-the-box, authorization plugins, localization capabilities, and more. The framework has numerous navigation options and API functions which can be used to tailor the app according to your needs. With the newest feature we offer - [Luigi Container](https://docs.luigi-project.io/docs/luigi-container) - you can also easily integrate micro frontends in your app with minimal setup involved. Unlike other frameworks like Single-SPA and Piral, Luigi follows the SAP Fiori design guidelines and provides a consistent and flexible way to build micro frontends that can work harmoniously with SAP applications.

* Use Cases - As mentioned previously, Luigi was developed with large-scale business applications in mind, and it's especially suitable for SAP applications. One of the main advantages of Luigi is that by using micro frontends, you can give large distributed teams ownership over their own components while maintaining a consistent user experience. Luigi is most suitable for splitting up large monolith frontends. You can find some of the businesses that have used Luigi on our [homepage](https://luigi-project.io/) in the section "Early adopters". 
 
* Drawbacks and Limitations - Like any other framework, Luigi also has some limitations depending on your use case. For example, if you are not coming from the SAP ecosystem, you may want to replace the default Fiori navigation with your own styling. (You can find some examples of how to do that under the[ FAQ question](https://docs.luigi-project.io/docs/faq?section=ui-questions) "I don't want to use the default Fiori Fundamentals style. How can I style Luigi differently?") Also, Luigi has a smaller community than frameworks like Single-SPA and Piral, but as it is open-source, it's possible to contribute any missing features that you want to see.
